### ISSUE TITLE
str/bytes TypeError in setDevelProject

### ISSUE THREAD

**Issue by mcepl** [openSUSE/osc#1008]
Using osc from master of the github (f4626dd91ff144850fc326f90669c29812777f8f):

```
stitny~$ osc maintainer --group=python-maintainer -D devel:langauges:python
Traceback (most recent call last):
  File "/home/matej/.bin/osc", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/home/matej/archiv/knihovna/repos/osc/osc-wrapper.py", line 45, in <module>
    r = babysitter.run(osccli)
  File "/home/matej/archiv/knihovna/repos/osc/osc/babysitter.py", line 67, in run
    return prg.main(argv)
  File "/home/matej/archiv/knihovna/repos/osc/osc/cmdln.py", line 347, in main
    return self.cmd(args)
  File "/home/matej/archiv/knihovna/repos/osc/osc/cmdln.py", line 370, in cmd
    retval = self.onecmd(argv)
  File "/home/matej/archiv/knihovna/repos/osc/osc/cmdln.py", line 504, in onecmd
    return self._dispatch_cmd(handler, argv)
  File "/home/matej/archiv/knihovna/repos/osc/osc/cmdln.py", line 1235, in _dispatch_cmd
    return handler(argv[0], opts, *args)
  File "/home/matej/archiv/knihovna/repos/osc/osc/commandline.py", line 8479, in do_maintainer
    setDevelProject(apiurl, prj, pac, opts.devel_project)
  File "/home/matej/archiv/knihovna/repos/osc/osc/core.py", line 7114, in setDevelProject
    path = (quote_plus(prj),) + (quote_plus(pac),)
  File "/usr/lib64/python3.8/urllib/parse.py", line 877, in quote_plus
    string = quote(string, safe + space, encoding, errors)
  File "/usr/lib64/python3.8/urllib/parse.py", line 861, in quote
    return quote_from_bytes(string, safe)
  File "/usr/lib64/python3.8/urllib/parse.py", line 886, in quote_from_bytes
    raise TypeError("quote_from_bytes() expected bytes")
TypeError: quote_from_bytes() expected bytes
stitny~$
```

**Comment by dmach** [comment]
@mcepl I thought this is going to be a simple fix, but it is quite opposite.
The code is about 12 years old and probably nobody used it for a while.
What exactly were you searching for?
What did you want to achieve by providing the devel project?
BTW, there's a typo in your command, I suppose it should be --group=python-maintainer**s** [1], but it doesn't work regardless the value.
[1] https://build.opensuse.org/groups/python-maintainers

**Comment by dmach** [comment]
I believe this should never occur again because the code was rewritten to use an object wrapper for manipulating the XML and it handles it consistently.
While looking into this I triggered a crash when running the command from a directory that is not a working copy and project/package are not specified on the command line. This is fixed in #1545
